File: server/FitMate.Core/JsonModels/AdminAI/UnsupportedRequestQueryRequest.cs
using System.ComponentModel.DataAnnotations;
using FitMate.Core.JsonModels.Common;
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.AdminAI;

public class UnsupportedRequestQueryRequest : PagedRequest
{
    [StringLength(200)]
    public string? Search { get; set; }

    [StringLength(100)]
    public string? Category { get; set; }

    public UnsupportedRequestStatus? Status { get; set; }
}

File: client/src/types/JsonModels/AdminAI/UnsupportedRequestQueryRequest.ts
import type { JsonModels } from "../../backend";
import type { UnsupportedRequestStatus } from "../Enums/UnsupportedRequestStatus";

export interface UnsupportedRequestQueryRequest extends JsonModels.Common.PagedRequest
{
	search?: string;
	category?: string;
	status?: UnsupportedRequestStatus;
}
